api/src/main/kotlin/xtdb/api/TransactionKey.kt

```diff
@@ -1,12 +1,16 @@
 package xtdb.api
 
 import clojure.lang.*
+import kotlinx.serialization.Contextual
+import kotlinx.serialization.Serializable
 import java.time.Instant
 
 private val TX_ID_KEY: Keyword = Keyword.intern("tx-id")
 private val SYSTEM_TIME_KEY: Keyword = Keyword.intern("system-time")
 
-data class TransactionKey(val txId: Long, val systemTime: Instant) : Comparable<TransactionKey>, ILookup, Seqable {
+
+@Serializable
+data class TransactionKey(val txId: Long, @Contextual val systemTime: Instant) : Comparable<TransactionKey>, ILookup, Seqable {
```

Comment:
I would also like to use `@Serialzable(AnySerde::class)` to encode Instant as a JsonLD type here, but kotlin doesn't like that `systemTime` can't be null.

Reply:
as discussed, even in JSON-LD we can just have Instant as a string where it's unambiguous